envoy/http/filter.h

```diff
@@ -331,10 +331,10 @@ class StreamFilterCallbacks {
   virtual ~StreamFilterCallbacks() = default;
 
   /**
-   * @return OptRef<const Network::Connection> the downstream connection, or nullptr if there is
+   * @return OptRef<Network::Connection> the downstream connection, or nullptr if there is
    * none.
    */
-  virtual OptRef<const Network::Connection> connection() PURE;
+  virtual OptRef<Network::Connection> connection() PURE;
```

Comment:
I am not sure if exposing non-const connection to the L7 filters is a good choice or not. I prefer leave this method as that and create a new `ConstStreamInfoWrapper` to expose the connection level stream info.

It's no doubt this will bring some redundant code. But it will reduce unnecessary interface change to the core code.

Or we can do a `const_cast` in the Lua filter. It's acceptable for me to use the non-const connection only in the Lua filter (untill we find better solution).

Reply:
Oh, I'd be very happy to use the const_cast solution, it's much more targeted than this mess is.

I'll do a const_cast update today or tomorrow, gleefully, and then explore how ugly the ConstStreamInfoWrapper change would be. (I'm thinking of a replacing StreamInfoWrapper with ConstStreamInfoWrapper, MutableStreamInfoWrapper, and StreamInfoWrapperBase, so we can try to avoid most of the duplication.)